sl_functionality.cc

```diff
@@ -280,15 +280,15 @@ void ServiceLayerFunctionality::broadcast_chirp(const std::string& username,
       client_->put(follower_key, monitor_serial);
     }
   }
-  std::vector<std::string> hashtags = chirp_hashtag_check(chirp);
-  for (int i = 0; i < hashtags.size(); i++) {
-    stream_chirp(chirp, hashtags[i]);
+  std::set<std::string> hashtags = chirp_hashtag_check(chirp);
+  for (std::string hash : hashtags) {
```

Comment:
Minor: I would use the noun rather than adjective (i.e., the variable should be "tags" not "hash").  Also, const reference.